pkg/kudoctl/resources/install/operator.go

```diff
@@ -143,6 +143,7 @@ func updateKudoOperatorTaskPackageNames(
 			for _, dependency := range dependencies {
 				if tasks[i].Spec.KudoOperatorTaskSpec.Package == dependency.PackageName {
 					tasks[i].Spec.KudoOperatorTaskSpec.Package = dependency.Operator.Name
+					tasks[i].Spec.KudoOperatorTaskSpec.OperatorVersion = dependency.OperatorVersion.Spec.Version
```

Comment:
@nfnt this is for cases when the user didn't specify the `KudoOperatorTaskSpec.OperatorVersion` and it was resolved to the latest one. the field has to be populated by the CLI

Reply:
Good catch! I wasn't aware of the implications of this field being optional. In this case, the same must be done for `KudoOperatorTaskSpec.AppVersion` if the dependency has a non-empty app version.

Reply:
Agree about the method name but the `AppVersion` is not needed on the server-side so 🤷